File: testdata/fixtures/ts/sample/greet.ts
import { readFile } from "fs";

export type User = {
  id: string;
  name: string;
};

export function greet(user: User): string {
  readFile("x");
  return format(user.name);
}

export const format = (name: string): string => {
  return name.toUpperCase();
};

export class Greeter {
  say(name: string): string {
    return format(name);
  }
}

File: testdata/fixtures/go/sample/greet.go
package main

import "fmt"

// helper prints a short greeting used by Greeter.
func helper() {
	fmt.Println("hi")
}

type Greeter struct{}

// Greet prints a personalized hello.
func (g *Greeter) Greet(name string) {
	helper()
	fmt.Printf("hello %s\n", name)
}

func main() {
	g := Greeter{}
	g.Greet("world")
	helper()
}
